crypto/BRCryptoHash.c

```diff
@@ -80,18 +80,27 @@ cryptoHashEqual (BRCryptoHash h1, BRCryptoHash h2) {
             : CRYPTO_FALSE);
 }
 
+static char *
+_cryptoHashAddPrefix (char *hash) {
+    char *result = malloc (strlen (hash) + 3);
+    strcpy (result, "0x");
+    strcat (result, hash);
+    free (hash);
```

Comment:
This (the free) seems like a scary side-effect to roll into the function.

Reply:
That is me obsessing on tail calls.  I pushed a change that at least makes it more explicit.